internal/compiler/passes/resolving.rs

```diff
@@ -156,6 +156,13 @@ enum LookupPhase {
     ResolvingTwoWayBindings,
 }
 
+pub fn resolve_at_gradient_with_ctx(
```

Comment:
This is super ugly!

I do want to use this code though: No use repeating it after all.

Reply:
why don't you make expression::from_at_gradient pub?

Reply:
Is that OK with you?

The implementation is still in the resolving pass, should it stay there?

Reply:
That's Ok for me. This is already the case for other functions such as Expression::from_binding_expression_node

In retrospect, is was maybe not ideal to put a `impl Expression` like that in passes/resolving.rs , maybe the whole thing could be moved around. But for now i think you can just use it.

Reply:
Done.